File: GXpert/GXpert.Web/Modules/Playlist/PlayList/PlayListColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Playlist.Columns;

[ColumnsScript("Playlist.PlayList")]
[BasedOnRow(typeof(PlayListRow), CheckNames = true)]
public class PlayListColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public string ClassTitle { get; set; }
    public string SubjectTitle { get; set; }
    public string TeacherPrn { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Playlist/PlayListColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PlayListRow } from "./PlayListRow";

export interface PlayListColumns {
    Id: Column<PlayListRow>;
    Title: Column<PlayListRow>;
    Description: Column<PlayListRow>;
    ClassTitle: Column<PlayListRow>;
    SubjectTitle: Column<PlayListRow>;
    TeacherPrn: Column<PlayListRow>;
}

export class PlayListColumns extends ColumnsBase<PlayListRow> {
    static readonly columnsKey = 'Playlist.PlayList';
    static readonly Fields = fieldsProxy<PlayListColumns>();
}